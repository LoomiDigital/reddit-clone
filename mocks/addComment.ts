import { MockedResponse } from "@apollo/client/testing";
import { AddCommentDocument } from "@d20/generated/graphql";

export const mockAddComment: MockedResponse = {
  request: {
    query: AddCommentDocument,
    variables: {
      post_id: 1,
      text: "This is a new comment",
      username: "Buck",
    },
  },
  result: {
    data: {
      addComment: {
        id: 3,
        text: "This is a new comment",
        post_id: 1,
        created_at: "2023-05-08T18:10:10.831966Z",
        username: "Buck",
      },
    },
  },
};
