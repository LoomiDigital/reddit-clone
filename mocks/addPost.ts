import { MockedResponse } from "@apollo/client/testing";
import { AddPostDocument } from "@d20/generated/graphql";

export const mockAddPost: MockedResponse = {
  request: {
    query: AddPostDocument,
    variables: {
      title: "This is a new post",
      body: "A brand new post!",
      image: "https://via.placeholder.com/150",
      username: "Buck",
      subreddit_id: 1,
      subreddit_topic: "testsubreddit",
    },
  },
  result: {
    data: {
      insertPost: {
        __typename: "Post",
        id: 1,
        title: "This is a new post",
        body: "A brand new post!",
        image: "https://via.placeholder.com/150",
        username: "Buck",
        subreddit_id: 1,
        subreddit_topic: "testsubreddit",
        created_at: "2021-08-15T20:00:00.000Z",
        votes: [
          {
            __typename: "Vote",
            id: -1,
            post_id: 1,
            username: "Buck",
            upvote: true,
          },
        ],
      },
    },
  },
};
