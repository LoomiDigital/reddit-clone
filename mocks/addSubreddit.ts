import { MockedResponse } from "@apollo/client/testing";
import { AddSubredditDocument } from "@d20/generated/graphql";

export const mockAddSubreddit: MockedResponse = {
  request: {
    query: AddSubredditDocument,
    variables: {
      topic: "testsubreddit",
    },
  },
  result: {
    data: {
      insertSubreddit: {
        __typename: "Subreddit",
        id: 1,
        post_id: 1,
        topic: "testsubreddit",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
  },
};
