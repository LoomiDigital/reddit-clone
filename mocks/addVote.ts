import { MockedResponse } from "@apollo/client/testing";
import { AddVoteDocument } from "@d20/generated/graphql";

export const mockAddVote: MockedResponse = {
  request: {
    query: AddVoteDocument,
    variables: {
      post_id: 1,
      username: "Buck",
      upvote: true,
    },
  },
  result: {
    data: {
      addVote: {
        __typename: "Vote",
        post_id: 1,
        upvote: true,
        username: "Buck",
      },
    },
  },
};
