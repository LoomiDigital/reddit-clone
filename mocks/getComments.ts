import { MockedResponse } from "@apollo/client/testing";
import {
  GetCommentsByPostIdDocument,
  GetPostsDocument,
  GetPostsQuery,
  PostConnection,
} from "@d20/generated/graphql";

export const mockCommentsResponse: MockedResponse = {
  request: {
    query: GetCommentsByPostIdDocument,
    variables: {
      post_id: 1,
    },
  },
  result: {
    data: {
      commentsByPostId: [
        {
          id: 1,
          text: "a comment",
          post_id: 1,
          created_at: "2023-05-08T18:10:10.831966Z",
          username: "barabis",
        },
        {
          id: 2,
          text: "another comment",
          post_id: 1,
          created_at: "2023-05-08T18:10:10.831966Z",
          username: "barabis",
        },
      ],
    },
  },
};

export const mockPosts: GetPostsQuery = {
  posts: {
    __typename: "PostConnection",
    edges: [
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 1,
          title: "Test Post",
          body: "Test Content",
          image: "http://placehold.it/300x300",
          username: "Buck",
          votes: [
            {
              upvote: true,
              username: "Buck",
              post_id: 1,
            },
          ],
          subreddit_id: 1,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 2,
          title: "Test Post",
          body: "Test Content",
          username: "Buck",
          image: "http://placehold.it/300x300",
          votes: [
            {
              upvote: true,
              username: "Buck",
              post_id: 2,
            },
          ],
          subreddit_id: 2,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
    ],
    pageInfo: {
      __typename: "PageInfo",
      hasNextPage: false,
      endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
    },
  },
};

export const mockPostConnection: PostConnection = {
  __typename: "PostConnection",
  edges: [
    {
      __typename: "PostEdge",
      cursor: "YXJyYXljb25uZWN0aW9uOjA=",
      node: {
        __typename: "Post",
        id: 1,
        title: "Test Post",
        body: "Test Content",
        image: "http://placehold.it/300x300",
        username: "Buck",
        votes: [
          {
            upvote: true,
            username: "Buck",
            post_id: 1,
          },
        ],
        subreddit_id: 1,
        subreddit_topic: "test",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
    {
      __typename: "PostEdge",
      cursor: "YXJyYXljb25uZWN0aW9uOjA=",
      node: {
        __typename: "Post",
        id: 2,
        title: "Test Post",
        body: "Test Content",
        username: "Buck",
        image: "http://placehold.it/300x300",
        votes: [
          {
            upvote: true,
            username: "Buck",
            post_id: 2,
          },
        ],
        subreddit_id: 2,
        subreddit_topic: "test",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
  ],
  pageInfo: {
    __typename: "PageInfo",
    hasPreviousPage: false,
    hasNextPage: false,
    startCursor: "YXJyYXljb25uZWN0aW9uOjA=",
    endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
  },
};
