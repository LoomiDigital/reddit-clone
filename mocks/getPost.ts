import { MockedResponse } from "@apollo/client/testing";
import {
  GetPostDocument,
  PostAttributesFragment,
} from "@d20/generated/graphql";

export const mockPostResponse: MockedResponse = {
  request: {
    query: GetPostDocument,
    variables: {
      id: 1,
    },
  },
  result: {
    data: {
      getPost: {
        node: {
          __typename: "Post",
          id: 1,
          title: "Test Post",
          body: "Test Content",
          image: "http://placehold.it/300x300",
          username: "Buck",
          votes: [
            {
              __typename: "Vote",
              upvote: true,
              username: "Buck",
              post_id: 1,
            },
          ],
          subreddit_id: 1,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
    },
  },
};

export const mockPost: PostAttributesFragment = {
  __typename: "Post",
  id: 1,
  title: "Test Post",
  body: "Test Content",
  image: "http://placehold.it/300x300",
  username: "Buck",
  votes: [
    {
      __typename: "Vote",
      upvote: false,
      username: "Buck",
      post_id: 1,
    },
  ],
  subreddit_id: 1,
  subreddit_topic: "test",
  created_at: "2021-08-15T20:00:00.000Z",
};

export const mockPostUpvoted: PostAttributesFragment = {
  __typename: "Post",
  id: 1,
  title: "Test Post",
  body: "Test Content",
  image: "http://placehold.it/300x300",
  username: "Buck",
  votes: [
    {
      __typename: "Vote",
      upvote: true,
      username: "Buck",
      post_id: 1,
    },
  ],
  subreddit_id: 1,
  subreddit_topic: "test",
  created_at: "2021-08-15T20:00:00.000Z",
};
