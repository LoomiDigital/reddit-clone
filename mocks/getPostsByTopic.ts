import { MockedResponse } from "@apollo/client/testing";

import {
  GetPostsByTopicDocument,
  GetPostsByTopicQuery,
} from "@d20/generated/graphql";

export const mockPostsByTopicResponse: MockedResponse = {
  request: {
    query: GetPostsByTopicDocument,
    variables: {
      first: 10,
    },
  },
  result: {
    data: {
      postsByTopic: {
        __typename: "PostConnection",
        edges: [
          {
            __typename: "PostEdge",
            cursor: "YXJyYXljb25uZWN0aW9uOjA=",
            node: {
              __typename: "Post",
              id: 1,
              title: "Test Post",
              body: "Test Content",
              image: "http://placehold.it/300x300",
              username: "Buck",
              votes: [
                {
                  __typename: "Vote",
                  upvote: true,
                  username: "Buck",
                  post_id: 1,
                },
              ],
              subreddit_id: 1,
              subreddit_topic: "test",
              created_at: "2021-08-15T20:00:00.000Z",
            },
          },
          {
            __typename: "PostEdge",
            cursor: "YXJyYXljb25uZWN0aW9uOjA=",
            node: {
              __typename: "Post",
              id: 2,
              title: "Test Post",
              body: "Test Content",
              username: "Buck",
              image: "http://placehold.it/300x300",
              votes: [
                {
                  __typename: "Vote",
                  upvote: true,
                  username: "Buck",
                  post_id: 2,
                },
              ],
              subreddit_id: 2,
              subreddit_topic: "test",
              created_at: "2021-08-15T20:00:00.000Z",
            },
          },
        ],
        pageInfo: {
          __typename: "PageInfo",
          hasNextPage: false,
          endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
        },
      },
    },
  },
};

export const mockPostsByTopic: GetPostsByTopicQuery = {
  postsByTopic: {
    __typename: "PostConnection",
    edges: [
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 1,
          title: "Test Post",
          body: "Test Content",
          image: "http://placehold.it/300x300",
          username: "Buck",
          votes: [
            {
              __typename: "Vote",
              upvote: true,
              username: "Buck",
              post_id: 1,
            },
          ],
          subreddit_id: 1,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
      {
        __typename: "PostEdge",
        cursor: "YXJyYXljb25uZWN0aW9uOjA=",
        node: {
          __typename: "Post",
          id: 2,
          title: "Test Post",
          body: "Test Content",
          username: "Buck",
          image: "http://placehold.it/300x300",
          votes: [
            {
              __typename: "Vote",
              upvote: true,
              username: "Buck",
              post_id: 2,
            },
          ],
          subreddit_id: 2,
          subreddit_topic: "test",
          created_at: "2021-08-15T20:00:00.000Z",
        },
      },
    ],
    pageInfo: {
      __typename: "PageInfo",
      hasNextPage: false,
      endCursor: "YXJyYXljb25uZWN0aW9uOjA=",
    },
  },
};

export const mockUseGetLazyPostsByTopicReturn = {
  posts: mockPostsByTopic.postsByTopic?.edges,
  loading: false,
  hasNextPage: true,
  sentryRef: jest.fn(),
};
