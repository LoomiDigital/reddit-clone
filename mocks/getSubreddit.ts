import { MockedResponse } from "@apollo/client/testing";
import {
  GetSubredditByTopicDocument,
  GetSubredditByTopicQuery,
  SubredditAttributesFragment,
} from "@d20/generated/graphql";

export const mockGetSubredditResponse: MockedResponse = {
  request: {
    query: GetSubredditByTopicDocument,
    variables: {
      topic: "testsubreddit",
    },
  },
  result: {
    data: {
      getSubredditByTopic: {
        __typename: "Subreddit",
        id: 1,
        topic: "testsubreddit",
        created_at: "2021-08-15T20:00:00.000Z",
      },
    },
  },
};

export const mockNoSubredditResponse: MockedResponse = {
  request: {
    query: GetSubredditByTopicDocument,
    variables: {
      topic: "testsubreddit",
    },
  },
  result: {
    data: {
      getSubredditByTopic: null,
    },
  },
};

export const mockSubreddit: SubredditAttributesFragment = {
  id: 1,
  topic: "testsubreddit",
  created_at: "2021-08-15T20:00:00.000Z",
};

export const mockSubredditResponse: GetSubredditByTopicQuery = {
  getSubredditByTopic: {
    __typename: "Subreddit",
    ...mockSubreddit,
  },
};
