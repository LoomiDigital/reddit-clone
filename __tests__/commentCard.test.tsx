import { MockedProvider } from "@apollo/client/testing";
import { SessionProvider } from "next-auth/react";
import { render, screen } from "@testing-library/react";
import { mockPostResponse } from "@d20/mocks/getPost";
import { mockComment, mockCommentsResponse } from "@d20/mocks/getComments";

import CommentCard from "@d20/components/CommentCard";

describe("CommentCard component", () => {
  it("renders the comment correctly", async () => {
    render(
      <MockedProvider mocks={[mockPostResponse, mockCommentsResponse]}>
        <SessionProvider
          session={{
            expires: "2021-10-10",
            user: {
              name: "Buck",
              expires: "2021-10-10",
              email: "user@test.com",
              address: "123 Fake St",
              image: "https://via.placeholder.com/150",
            },
          }}
        >
          <CommentCard comment={mockComment} />
        </SessionProvider>
      </MockedProvider>
    );

    expect(await screen.findByText("An amazing comment")).toBeInTheDocument();
    expect(await screen.findByText("aUser")).toBeInTheDocument();
  });
});
