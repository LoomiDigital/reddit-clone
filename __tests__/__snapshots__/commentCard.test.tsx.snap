// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentCard component should render correctly 1`] = `
<div
  className="relative flex items-center space-x-2 space-y-10"
>
  <hr
    className="absolute left-7 top-16 z-0 h-16 border"
  />
  <div
    className="z-50"
  >
    <div
      className="relative h-10 w-10 overflow-hidden rounded-full border-gray-300 bg-white undefined"
    >
      <img
        alt="Buck"
        data-nimg="fill"
        decoding="async"
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="https://avatars.dicebear.com/api/open-peeps/aUser.svg"
        style={
          {
            "bottom": 0,
            "color": "transparent",
            "height": "100%",
            "left": 0,
            "objectFit": undefined,
            "objectPosition": undefined,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": "100%",
          }
        }
      />
    </div>
  </div>
  <div
    className="flex flex-col"
  >
    <p
      className="py-2 text-xs text-gray-400"
    >
      <span
        className="font-semibold text-gray-600"
      >
        aUser
      </span>
       
      â€¢ 
    </p>
    <p
      className="text-sm"
    >
      An amazing comment
    </p>
  </div>
</div>
`;
